transport/http/outbound.go

```diff
@@ -132,21 +153,64 @@ func (t *Transport) NewOutbound(chooser peer.Chooser, opts ...OutboundOption) *O
 		opt(o)
 	}
 
-	client := t.client
-	if o.tlsConfig != nil {
-		client = createTLSClient(o)
-		// Create a copy of the url template to avoid scheme changes impacting
-		// other outbounds as the base url template is shared across http
-		// outbounds.
-		ut := *o.urlTemplate
-		ut.Scheme = "https"
-		o.urlTemplate = &ut
+	var client *http.Client
+	if o.useHTTP2 {
+		client = createHTTP2Client(o)
+	} else {
+		client = t.client
+		if o.tlsConfig != nil {
+			client = createTLSClient(o)
+			// Create a copy of the url template to avoid scheme changes impacting
+			// other outbounds as the base url template is shared across http
+			// outbounds.
+			ut := *o.urlTemplate
+			ut.Scheme = "https"
+			o.urlTemplate = &ut
+		}
 	}
 	o.client = client
 	o.sender = &transportSender{Client: client}
 	return o
 }
 
+func createHTTP2Client(o *Outbound) *http.Client {
+	if o.tlsConfig != nil {
+		return createHTTP2TLSClient(o)
+	}
+
+	// check if transport is already http2 transport
+	if _, ok := o.transport.client.Transport.(*http2.Transport); ok {
+		return o.transport.client
+	}
+
+	// get default http1 transport from the client and use it to create http2 transport
+	http1Transport, ok := o.transport.client.Transport.(*http.Transport)
+	if !ok {
+		// This should not happen as default yarpc http.Client uses
+		// http.Transport and it's not configurable by the user.
+		panic(fmt.Sprintf("failed to create http tls client, provided http.Client transport type %T is not *http.Transport", o.transport.client.Transport))
```

Comment:
Is the panic message correct? (It says about tls client, though it doesn't seem like a creation of the tls client)

Reply:
oops, nice, thanks!